src/SFML/Window/EGLCheck.hpp

```diff
@@ -54,20 +59,28 @@ bool eglCheckError(const std::filesystem::path& file, unsigned int line, std::st
 
 // In debug mode, perform a test on every EGL call
 // The lamdba allows us to call eglCheck as an expression and acts as a single statement perfect for if/else statements
-#define eglCheck(...)                                                                             \
-    [](auto&& eglCheckInternalFunction)                                                           \
-    {                                                                                             \
-        if constexpr (!std::is_void_v<decltype(eglCheckInternalFunction())>)                      \
-        {                                                                                         \
-            const auto eglCheckInternalReturnValue = eglCheckInternalFunction();                  \
-            sf::priv::eglCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-            return eglCheckInternalReturnValue;                                                   \
-        }                                                                                         \
-        else                                                                                      \
-        {                                                                                         \
-            eglCheckInternalFunction();                                                           \
-            sf::priv::eglCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-        }                                                                                         \
+#define eglCheck(...)                                                                                     \
+    [](auto&& eglCheckInternalFunction)                                                                   \
+    {                                                                                                     \
+        if (const EGLint error = eglGetError(); error != EGL_SUCCESS)                                     \
```

Comment:
choose a longer name like eglCheckInternalError to not cause issues with shadowing.